semgrep-core/src/analyzing/Constant_propagation.ml

```diff
@@ -672,18 +672,9 @@ let propagate_dataflow lang ast =
       let xs =
         AST_to_IL.stmt lang (G.Block (Parse_info.unsafe_fake_bracket ast) |> G.s)
       in
-      let flow = CFG_build.cfg_of_stmts xs in
+      let flow = CFG_build.cfg_of_stmts lang xs in
       propagate_dataflow_one_function lang [] flow
   | _ ->
-      let v =
-        V.mk_visitor
-          {
-            V.default_visitor with
-            V.kfunction_definition =
-              (fun (_k, _) def ->
-                let inputs, xs = AST_to_IL.function_definition lang def in
-                let flow = CFG_build.cfg_of_stmts xs in
-                propagate_dataflow_one_function lang inputs flow);
-          }
-      in
-      v (Pr ast)
+      let xs = AST_to_IL.stmt lang (G.stmt1 ast) in
+      let flow = CFG_build.cfg_of_stmts lang xs in
+      propagate_dataflow_one_function lang [] flow
```

Comment:
nice! Maybe rename propagate_dataflow_one_function to propagate_dataflow_program?

Reply:
With the way that it's being exposed for DeepSemgrep, I think it may be best to keep the name the same.

One side note is that both before and currently, `_inputs` is being ignored completely. I'm not sure if that's supposed to be used, or if DeemSemgrep is assuming that it is used...